### ISSUE TITLE
Possible Inconsistency on Prediction Notebook

### ISSUE THREAD

**Issue by leobasilio** [mkhushi/code-with-paper#2]
Hi. I've been studying your Forex prediction paper for a couple of days now and, first of all, thank you very much for publishing the code too. I've been learning a lot from it, but I think there might be a problem with the prediction notebook. As far as I understand, the features extracted by the first layers are collected in the `feature` tensor. Then they're converted to a numpy array, concatenated with the technical indicators and the whole thing is used as input for LightGBM. My point is that the concatenation of `x1` and `x2` rely on the fact that both are equally ordered. I mean, every time window has the same index on both arrays.

On the other hand, the `DataLoader` used to iterate the batched data has `shuffle=True`.

```python
data_loader_FinalM5train = DataLoader(FinaldatasetM5_train, batch_size=BATCH_SIZE,shuffle=True)
```

So every epoch the batches are iterated in a random order. Therefore, the `feature` tensor is also shuffled, breaking the relationship between `x1` and `x2`. It doesn't seem right. :thinking: 

Did I miss something?

Thank you.

**Comment by leobasilio** [comment]
Never mind. I just realized there's a commented block that would generate the `feature` tensor in the right order. :grimacing:
